lib/primitive/lib/Cardano/Wallet/Primitive/Types/TokenMap.hs

```diff
@@ -690,18 +643,7 @@ removeQuantity m asset = setQuantity m asset TokenQuantity.zero
 -- | Get the largest quantity from this map.
 --
 maximumQuantity :: TokenMap -> TokenQuantity
-maximumQuantity =
-    Map.foldl' (\a -> Map.foldr findMaximum a . NEMap.toMap) zero . unTokenMap
-  where
-    zero :: TokenQuantity
-    zero = TokenQuantity 0
-
-    findMaximum :: TokenQuantity -> TokenQuantity -> TokenQuantity
-    findMaximum challenger champion
-        | challenger > champion =
-            challenger
-        | otherwise =
-            champion
+maximumQuantity = F.foldl' (F.foldl' max) mempty . unTokenMap
```

Comment:
`F.foldl' max == F.maximum` ?
https://hackage.haskell.org/package/base-4.18.0.0/docs/src/Data.Foldable.html#maximum

Reply:
> `F.foldl' max == F.maximum` ? https://hackage.haskell.org/package/base-4.18.0.0/docs/src/Data.Foldable.html#maximum

Thanks @Unisay! Though I can't see a way to make it work with `F.maximum`.

As I understand it, `maximum` will fail with a run-time error if applied to an empty structure. But we want the `maximumQuantity` function:
- to be total
- to return `mempty = TokenQuantity.zero` if applied to an empty `TokenMap`.

I've annotated the existing function with types, to make it a bit clearer:
```hs
maximumQuantity :: TokenMap -> TokenQuantity
maximumQuantity = F.foldl' acc (mempty :: TokenQuantity) . unTokenMap
  where
    acc :: TokenQuantity -> MonoidMap TokenName TokenQuantity -> TokenQuantity
    acc currentMax innerMap =
        F.foldl'
            (max :: TokenQuantity -> TokenQuantity -> TokenQuantity)
            currentMax
            innerMap
```